ee/localserver/request-query.go

```diff
@@ -19,48 +21,100 @@ func (ls *localServer) requestQueryHanlderFunc(w http.ResponseWriter, r *http.Re
 		return
 	}
 
-	var body map[string]any
+	var body map[string]string
 	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
 		sendClientError(w, fmt.Sprintf("error unmarshaling request body: %s", err))
 		return
 	}
 
-	queryRaw, ok := body["query"]
-	if !ok {
+	query, ok := body["query"]
+	if !ok || query == "" {
 		sendClientError(w, "no query key found in request body json")
 		return
 	}
 
-	query, ok := queryRaw.(string)
-	if !ok {
-		sendClientError(w, fmt.Sprintf("query value not a string: %s", query))
+	var results []map[string]string
+	var err error
+	backoff.WaitFor(func() error {
+		results, err = ls.querier.Query(query)
+		return err
+	}, 1*time.Second, 250*time.Millisecond)
+
+	if err != nil {
+		sendClientError(w, fmt.Sprintf("error executing query: %s", err))
 		return
 	}
 
-	if query == "" {
-		sendClientError(w, "empty query")
+	jsonBytes, err := json.Marshal(results)
+	if err != nil {
+		sendClientError(w, fmt.Sprintf("error marshalling results to json: %s", err))
 		return
 	}
 
+	w.Write(jsonBytes)
+}
+func (ls *localServer) requestScheduledQueryHandler() http.Handler {
+	return http.HandlerFunc(ls.requestScheduledQueryHandlerFunc)
+}
+
+// requestScheduledQueryHandlerFunc uses the name field in the request body to look up
+// an existing osquery scheduled query execute it, returning the results.
+func (ls *localServer) requestScheduledQueryHandlerFunc(w http.ResponseWriter, r *http.Request) {
+	// The driver behind this is that the JS bridge has to use GET requests passing the query (in a nacl box) as a URL parameter.
+	// This means there is a limit on the size of the query. This endpoint is intended to be a work around for that. It ought to work like this:
+	//
+	// 1. K2 looks up the name of a osquery scheduled query it want to be run. The same query should be available to launcher in the osquery_schedule table.
+	// 2. K2 calls `/scheduledquery`, with body `{ "name": "name_of_scheduled_query" }`
```

Comment:
Do you think this should be a different endpoint, or a different parameter to the same endpoint? (Either seems fine)

Reply:
I prefer a different endpoint .... seems easier that splitting logic path of single endpoint